tests/components/sunweg/test_sensor.py

```diff
@@ -0,0 +1,185 @@
+"""Tests for the Sun WEG sensor."""
+
+from freezegun.api import FrozenDateTimeFactory
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.sunweg.const import DOMAIN
+from homeassistant.components.sunweg.coordinator import SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .common import SUNWEG_MOCK_ENTRY
+
+from tests.common import async_fire_time_changed
+
+
+async def test_sensor_total(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    freezer: FrozenDateTimeFactory,
+    api_fixture,
+    plant_fixture_alternative,
+) -> None:
+    """Test sensor type total."""
+    mock_entry = SUNWEG_MOCK_ENTRY
+    mock_entry.add_to_hass(hass)
+
+    assert await async_setup_component(hass, DOMAIN, mock_entry.data)
```

Comment:
Sunweg is a config entry based integration, that shouldn't start like this, but rather with `await hass.config_entries.setup(entry.entry_id)` (out the top of my head